add more detail / clarification to conflict-of-interest guidelines (#364)
cncf/sig-security: assessments/guide/security-reviewer.md

# Security reviewer

## Lead reviewer
The lead reviewer serves as the primary security assessment facilitator, often coordinating 
responses between the project and reviewers, ensuring communications are performed in a timely 

> fashion as described in the [guide](./).  The lead reviewer also responsible for coordinating 

Reviewer: If I recall, I've always done COI (or asked the chairs to).  I feel like pushing this to the ever-changing lead is a way to make this more uneven, which feels like the wrong thing to outsource especially since this is so little work...

> and recruiting additional reviewers as appropriate, reviewing and deliberating conflict of 
> interest statements, and managing the documentation updates (self-assessment and final 

Reviewer 2: ```suggestion
and recruiting additional reviewers as appropriate, and managing the documentation updates (self-assessment and 
```

> assessment) to the project's assessment folder.

Reviewer 2: ```suggestion
final assessment) to the project's assessment folder.
```

The conflicts of interest lie on a spectrum, and are classified into hard and soft conflicts:
* A hard conflict makes a reviewer ineligible to review a project.
* A soft conflict allows a reviewer to review a project, but not as a [project lead](./project-lead.md).
* It is not unusual for reviewers to have soft conflicts. The diversity of reviewers that are familiar with a project and can provide a deeper insight together with a fresh set of eyes is beneficial to the success of a security review.

All reviewers must provide a conflict declaration to indicate which hard or soft conflicts do, or do not exist 
when they volunteer to be a reviewer.  This is done by placing a comment on the issue associated with the 
assessment.
Should a conflict arise during the time of the assessment, reviewers should notify the lead security reviewer when 
they become aware of the potential conflict, so the new conflict may be consulted with the facilitator and/or chairs.

The lead security reviewer has the responsibility of ensuring a balanced review,
and as part of that before kicking off the assessment must:
* Check that all reviewers have conflict-of-interest declarations,

> * make their declaration of any potential conflict-of-interest (or lack thereof)
> * ask clarifying questions, if needed
> * assert that:
>   1. they have reviewed all conflict-of-interest declarations;
>   2. there are no hard-conflicts;
>   3. they believe that the review team is able to provide a balanced and fair assessment

Reviewer 2: ```suggestion
* Provide their own declaration of any potential conflict-of-interest (or lack thereof),
* Ask clarifying questions, if needed
* Assert that:
  1. They have reviewed all conflict-of-interest declarations from the reviewers;
  2. There are no hard-conflicts present without a waiver;
  3. They believe that the review team is able to provide a balanced and fair assessment
```

Update the above assertion if a new conflict-of-interest becomes known during
the course of the assessment.

The Security Assessment Facilitator or a SIG-Security chair must review the 
Lead Security Reviewer conflict-of-interest assertion.

> If there any conflicts are presented, then 2 SIG-Security chairs must approve

Reviewer 3: this includes soft conflicts right, this seems like it may be a potentially high overhead for chairs, since soft conflicts are likely to be common. How about making this discretionary to the security assessment facilitator

Reviewer 2: So the way this is written, means that either the Security Assessment Facilitator Or a chair must review the entire assertion.  assertions include both hard and soft conflicts. I'll adjust the "any conflicts" to be "any hard conflicts, or multiple significant soft conflicts" for the 2 chair approval.

> If there any conflicts are presented, then 2 SIG-Security chairs must approve

Reviewer 2: ```suggestion
If there any hard conflicts, or multiple significant soft conflicts are presented, then 2 SIG-Security chairs must approve
```

the security review team. Reasons for accepting and rejecting conflicts should
be documented.
In most cases, the existence of a hard conflict will prevent a SIG member from participating in the assessment for 
which their hard conflict exists.  Depending on the circumstances of the particular conflict, the assessment, and the 

> may project, the chairs and the assessment facilitator may determine if the hard conflict may be waived.  Should this 

Reviewer 2: ```suggestion
project, two chairs and the assessment facilitator may determine if the hard conflict may be waived.  Should this 
```

occur, the decision's justification will be documented to ensure it clearly depicts the circumstances for granting the 

> waiver.

Reviewer 3: I think the 2 chair requirement should perhaps be for this?

Reviewer 2: corrected the wording, somehow it was miswritten, let me know if this works @lumjjb

- SIG-Security chairs and assessment leads could intentionally or unintentionally limit the participation of an individual unfairly by asserting conflict of interest
- Security reviews being stalled while groups belabor on who should be allowed to participate

The conflicts of interest lie on a spectrum, and are classified into hard and soft conflicts:

> * Hard conflicts make a reviewer ineligible to review a project.

Reviewer: ```suggestion
* A hard conflict makes a reviewer ineligible to review a project.
```

* A soft conflict allows a reviewer to review a project, but not as a [project lead](./project-lead.md).

> * Soft conflicts are expected for some reviewers, and therefore are fine as long

Reviewer: ```suggestion
* In some cases, one or more reviewers may have a soft conflict.  This may be permissible so long
```
Change is moving plural to singular for clarity.

Author: There's been discussion that we really want someone on the review team that has experience using the project, yet that is a soft conflict, so I was trying to word this to make clear that we're actually *encouraging* at least one person with a soft conflict.  I didn't want to make a long section even longer, but maybe it would be good to just say that: 

> Typically, at least one reviewer will have a soft conflict (since the review will benefit from the perspective of someone experienced with the project).  This is permissible so long...

What do you think?

Reviewer 3: hmm, how about:

"It is not unusual for reviewers to have soft conflicts. The diversity of reviewers that are familiar with a project and can provide a deeper insight together with a fresh set of eyes is beneficial to the success of a security review."

Reviewer 2: ```suggestion
* It is not unusual for reviewers to have soft conflicts. The diversity of reviewers that are familiar with a project and can provide a deeper insight together with a fresh set of eyes is beneficial to the success of a security review.
```

>   as the review team includes enough different perspectives to ensure a

Reviewer 2: ```suggestion
```

>   balanced assessment.

Reviewer 2: ```suggestion
```

A reviewer with a any conflict or potential conflect is required to document the nature of the conflict when volunteering as a reviewer, or at any time as they
become aware of a potential conflict. It is the responsibility of the security 
reviewer to make known his/her own conflict of interests and ensure that

> their declaration has been reviewed and approved by lead security reviewer.

Reviewer: I think the lead being aware is good, but this does also seem to mildly conflict with the statement below about the facilitator / chairs reviewing the conflict of interest.  Do we want the lead assessor to have the final say here?  I'd prefer it was centralized across reviews.

Author: The above section is attempting to make clear that individual reviewers must make sure they document a conflict-of-interest if it happens after initial approval (which happened a while back via acquisition / merger) and its not enough to just comment on the issue, they have to also make sure the lead security reviewer acknowledges their note -- of course that may entail wider review depending on the nature of the conflict.  Not sure how to make that more clear...

> A reviewer with a any conflict or potential conflect is required to document the nature of the conflict when volunteering as a reviewer, or at any time as they
> become aware of a potential conflict. It is the responsibility of the security 
> reviewer to make known his/her own conflict of interests and ensure that
> their declaration has been reviewed and approved by lead security reviewer.

Reviewer 2: ```suggestion
All reviewers must provide a conflict declaration to indicate which hard or soft conflicts do, or do not exist 
when they volunteer to be a reviewer.  This is done by placing a comment on the issue associated with the 
assessment.
Should a conflict arise during the time of the assessment, reviewers should notify the lead security reviewer when 
they become aware of the potential conflict, so the new conflict may be consulted with the facilitator and/or chairs.
```

The lead security reviewer has the responsibility of ensuring a balanced review,
and as part of that before kicking off the assessment must:

> * check that all reviewers have conflict-of-interest declarations

Reviewer 3: consistency with bullets on top, caps or no caps

> * check that all reviewers have conflict-of-interest declarations

Reviewer 2: ```suggestion
* Check that all reviewers have conflict-of-interest declarations,
```

> * make their declaration of any potential conflict-of-interest (or lack thereof)

Reviewer 3: when reading these bullets, 'their'  and 'they' was a bit ambiguous (referring to reviewers or lead reviewers?) Since there was a mention in bullet 1 with "all reviewers"

* ask clarifying questions, if needed
* assert that:
  1. they have reviewed all conflict-of-interest declarations;
  2. there are no hard-conflicts;

>   4. they believe that the review team is able to provide a balanced and fair    assessment

Reviewer: ```suggestion
  3. they believe that the review team is able to provide a balanced and fair assessment
```

>   4. they believe that the review team is able to provide a balanced and fair    assessment

Reviewer 2: ```suggestion
* Ask clarifying questions, if needed,
* Assert that:
  1. They have reviewed all conflict-of-interest declarations;
  2. There are no hard-conflicts; and
  4. They believe that the review team is able to provide a balanced and fair assessment
```

Update the above assertion if a new conflict-of-interest becomes known during
the course of the assessment.

The Security Assessment Facilitator or a SIG-Security chair must review the 
Lead Security Reviewer conflict-of-interest assertion.

> If there any conflicts are presented, then 2 SIG-Security chairs must approve

Reviewer: This is for soft conflicts, I assume.  Can hard conflicts ever be waived?  Can a conflicted person still participate unofficially?

Author: It's hard to imagine that we would waive a hard conflict, I'm open to writing down that we would disallow chair approval of such.

I think we say elsewhere that any SIG member (or community member?) can participate... would it be helpful to make it explicit here?  I don't want to imply that there is some kind of "unofficial reviewer" role... just cuz that feels complicated.  Maybe someone can suggest words :)

the security review team. Reasons for accepting and rejecting conflicts should

> be documented.

Reviewer 2: ```suggestion
be documented.
In most cases, the existence of a hard conflict will prevent a SIG member from participating in the assessment for 
which their hard conflict exists.  Depending on the circumstances of the particular conflict, the assessment, and the 
may project, the chairs and the assessment facilitator may determine if the hard conflict may be waived.  Should this 
occur, the decision's justification will be documented to ensure it clearly depicts the circumstances for granting the 
waiver.
```
